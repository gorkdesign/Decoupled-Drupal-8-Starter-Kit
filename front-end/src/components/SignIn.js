import React from 'react'; 
import PropTypes from 'prop-types';

import 'styles/signin.css';

const signIn = props => {
  return (
    <div className="signin-container">    
      <div className="text-center">
        <form className="form-signin" onSubmit={ props.handleLogin }>
          <h1 className="h3 mb-3 font-weight-normal">Please sign in</h1>
          <input className="form-control" name="username" type='username' placeholder='Username' onChange={ props.handleInputChange } />
          <input className="form-control" name="password" type='password' placeholder='Password' onChange={ props.handleInputChange } />
          <div className="checkbox mb-3">
            <label>
              <input type="checkbox" value="remember-me" /> Remember me
            </label>
          </div>
          <button className="btn btn-lg btn-primary btn-block" type="submit">Sign in</button>
        </form>
        <p style={{color: 'red', display: props.isLoginFailed ? 'block' : 'none'}}> Credentials incorrect</p>
      </div>
    </div>
  );
}

signIn.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
  handleLogin: PropTypes.func.isRequired,
  isLoginFailed: PropTypes.bool.isRequired
}

export default signIn;