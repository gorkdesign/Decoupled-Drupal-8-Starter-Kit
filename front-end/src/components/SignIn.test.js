import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
// import renderer from 'react-test-renderer';

import SignIn from './SignIn';


describe('SignIn', () => {
  const initialPropsState = {
    handleInputChange: jest.fn(),
    handleLogin: jest.fn(),
    isLoginFailed: false,
  };

  let props;
  let mountedSignIn;
  const signIn = () => {
    if (!mountedSignIn) {
      mountedSignIn = mount(<SignIn {...props} />);
    }
    return mountedSignIn;
  };

  beforeEach(() => {
    props = initialPropsState;
    mountedSignIn = undefined;
  });

  describe('Snapshots', () => {
    const output = shallow(<SignIn {...initialPropsState} />);
    it('should render correctly', () => {
      expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('renders a email input', () => {
      expect(output.find('.username').length).toEqual(1);
    });

    it('renders a password input', () => {
      expect(output.find('.password').length).toEqual(1);
    });

    it('should respond to change event', () => {
      output.find('.username').simulate('change', { target: { name: 'username', value: 'test' } });
      expect(props.handleInputChange.mock.calls.length).toBe(1);
    });
  });

  it('always renders a div', () => {
    const divs = signIn().find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  it('always renders a `username`, `password`, and `submit` button', () => {
    const username = signIn().find('.username');
    expect(username.length).toBe(1);

    const password = signIn().find('.password');
    expect(password.length).toBe(1);

    const submit = signIn().find('.submit');
    expect(submit.length).toBe(1);
  });

  describe('SignIn differing props', () => {
    beforeEach(() => {
      props.isLoginFailed = true;
    });

    it('Displays an error message when `isLoginFailed` is passed in as true', () => {
      const isLoginFailed = signIn().find('.error');
      expect(isLoginFailed.length).toBe(1);
    });
  });
});
