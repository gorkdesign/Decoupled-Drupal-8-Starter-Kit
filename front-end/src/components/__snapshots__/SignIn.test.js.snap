// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render correctly 1`] = `
<div
  className="signin-container"
>
  <div
    className="text-center"
  >
    <form
      className="form-signin"
      onSubmit={[Function]}
    >
      <h1
        className="h3 mb-3 font-weight-normal"
      >
        Please sign in
      </h1>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="username"
      />
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <div
        className="checkbox mb-3"
      >
        <label>
          <input
            type="checkbox"
            value="remember-me"
          />
           Remember me
        </label>
      </div>
      <button
        className="btn btn-lg btn-primary btn-block"
        type="submit"
      >
        Sign in
      </button>
    </form>
    <p
      style={
        Object {
          "color": "red",
          "display": "none",
        }
      }
    >
       Credentials incorrect
    </p>
  </div>
</div>
`;
